include/wasm.hh

```diff
@@ -180,24 +180,37 @@ public:
 
 // Ownership
 
-template<class T> using own = std::unique_ptr<T>;
+template <typename T>
+class Destroyer {
+public:
+  void operator()(T *ptr) {
+    ptr->destroy();
+  }
+};
+
+template<class T> using own = std::unique_ptr<T, Destroyer<T>>;
 template<class T> using ownvec = vec<own<T>>;
 
 template<class T>
 auto make_own(T* x) -> own<T> { return own<T>(x); }
 
+template<class To, class From>
+auto own_cast(own<From> x) -> own<To> { return make_own<To>(x.release()); }
```

Comment:
```suggestion
auto own_cast(own<From> x) -> own<To> { return own<To>(x.release()); }
```

Reply:
Why does `make_own` exist? I noticed that wasm-v8.cc uses `own<T>` directly and mimicked that in my changes, but in wasm.hh I thought that making perhaps a pointer should go through the make function.

Is it supposed to be a parallel to `std::make_unique`? make_unique doesn't take a pointer, it takes the arguments that the T's c'tor would take and forwards them along. This wouldn't be useful to V8 because wasm-v8.cc uses `new(std::nothrow)` instead of regular `new`.

Reply:
I've gone ahead and removed `make_own`.

Reply:
`make_own` exists for the same reason that `make_pair` and friends exist in the std lib: to work around C++'s odd inability to infer template arguments for constructor invocations. So this would mostly be for convenience in user code, not necessarily the implementation.

Reply:
Okay, but in this case it might be confusing because it's different from make_unique while otherwise own<T> is an alias to unique_ptr<T>. I've put it back, but we still don't use it in wasm-v8.